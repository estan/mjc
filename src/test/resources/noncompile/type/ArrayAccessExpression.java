class ArrayAccessExpressionTest {
    public static void main(String[] args) {
        int i;
        int[] ia;
        boolean b;
        A cA;
        B cB;

        //i = ia[i];// OK!
        i = ia[ia]; // INVALID_INDEX_TYPE
        i = ia[b];  // INVALID_INDEX_TYPE
        i = ia[cA]; // INVALID_INDEX_TYPE

        i = i[i];    // NOT_ARRAY_TYPE
        i = b[i];    // NOT_ARRAY_TYPE
        i = cA[i];   // NOT_ARRAY_TYPE
    }
}
class A {}
class B {}
