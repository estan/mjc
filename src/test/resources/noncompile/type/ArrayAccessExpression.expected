INVALID_INDEX_TYPE
INVALID_INDEX_TYPE
INVALID_INDEX_TYPE
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
